lgc/patch/PatchBufferOp.cpp

```diff
@@ -421,11 +421,28 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
     Value *const pointer = getPointerOperandAsInst(callInst.getArgOperand(0));
 
     // Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
-    Value *const bufferLength = m_builder->CreateExtractElement(m_replacementMap[pointer].first, 2);
+    Value *const bufferDesc = m_replacementMap[pointer].first;
+    Value *numRecords = m_builder->CreateExtractElement(bufferDesc, 2);
+
+    // The result is 0 for a null buffer descriptor
+    if (m_pipelineState->getOptions().robustAccess) {
+      // Check if the buffer descriptor all bits 0
+      Value *const descIcmpVec = m_builder->CreateICmpEQ(bufferDesc, Constant::getNullValue(bufferDesc->getType()));
+      Value *isNullDesc = m_builder->CreateExtractElement(descIcmpVec, uint64_t(0));
+      for (unsigned elemIdx = 1; elemIdx != 4; ++elemIdx) {
```

Comment:
I think you should use targetInfo->getGpuProperty().descriptorSizeBuffer to replace 4

Reply:
Fixed.